@xen-orchestra/xapi/vm.js

```diff
@@ -343,7 +343,7 @@ class Vm {
     const vm = await this.getRecord('VM', vmRef)
 
     if (!bypassBlockedOperation && 'destroy' in vm.blocked_operations) {
-      throw new Error('destroy is blocked')
+      throw forbiddenOperation('destroy is blocked')
```

Comment:
`blocked_operations` is a map which associates an blocked action to a reason, maybe we should use this reason.

What do you think?

Reply:
Hm, maybe i misunderstand, but  `blocked_operations` is something like:
```js
{
...,
destroy: 'true'
}
```
I am not seeing any attached reason to the blocked operation.

Reply:
This is the reason attached to this blocked operation.

It just happen that this is what `xo-server` is using when protecting a VM, maybe we should use a better value so that we can differentiate protected VMs from other situations?

Reply:
Yes, i think it can be nice to have more explicit value as reason.

Reply:
Done on server's side.